pkg/blockbuilder/blockbuilder.go

```diff
@@ -441,17 +456,35 @@ func getGroupLag(ctx context.Context, admClient *kadm.Client, topic, group strin
 		return nil, err
 	}
 
-	// For now, if the group doesn't have any commits, we use startOffsets as the starting point.
-	if len(offsets) == 0 {
-		offsets = make(kadm.OffsetResponses)
-		for t, pt := range startOffsets.Offsets() {
-			resp := make(map[int32]kadm.OffsetResponse)
-			for _, o := range pt {
-				resp[o.Partition] = kadm.OffsetResponse{
-					Offset: o,
-				}
+	listOffsetsAfterMilli := sync.OnceValues(func() (kadm.ListedOffsets, error) {
+		return admClient.ListOffsetsAfterMilli(ctx, fallbackOffset, topic)
+	})
+	// If the group-partition in offsets doesn't have a commit, fall back depending on where fallbackOffset points at.
+	for topic, pt := range startOffsets.Offsets() {
+		for part, o := range pt {
+			if _, ok := offsets.Lookup(topic, part); ok {
+				continue
+			}
+			// If fallback is not a timestamp, it's the kafkaOffsetStart value.
+			if fallbackOffset < 0 {
+				offsets.Add(kadm.OffsetResponse{Offset: o})
+				continue
+			}
+			fallbackOffsets, err := listOffsetsAfterMilli()
+			if err != nil {
+				return nil, err
+			}
+			o, ok := fallbackOffsets.Lookup(topic, part)
+			if !ok {
+				// This should not happen.
+				return nil, fmt.Errorf("partition %d not found in fallback offsets for topic %s", part, topic)
```

Comment:
Can this error occur if the partition has no data?

Reply:
Here, we're inside the loop that iterates over topic-partitions from the resolved `startOffsets`. I.e. the partition here is known to have data.

Reply:
I've added a test for this and other cases. PHAL